### PR TITLE
Docs: Updates BGP CP for LB Class Support

### LINKED ISSUE #27056: Docs: Add LoadBalancerClass to BGP CP Service Announcements
Update the [service-announcements](https://docs.cilium.io/en/latest/network/bgp-control-plane/#service-announcements) docs for LoadBalancerClass support when https://github.com/cilium/cilium/pull/26905 merges.

### PR DESCRIPTION
**Pull request by danehans**
Adds [LB class](https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class) support to BGP CP docs.

Fixes: #27056

```release-note
docs: Update BGP control plane documentation with regards to LB class support and service announcements
```

### CODE REVIEW

**Comment by danehans** [comment]
cc @rastislavs

**Review by harsimran-pabla — APPROVED** [review]

File: Documentation/network/bgp-control-plane.rst [hunk 1]
```diff
@@ -364,6 +364,9 @@ with a dummy key and value can be used like:
          matchExpressions:
             - {key: somekey, operator: NotIn, values: ['never-used-value']}
 
+A service must have `loadBalancerClass <https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class>`__
+unspecified or set to "io.cilium/bgp-control-plane" to be announced by a policy.
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]: Nit: I'd rather have pre-formatted text to mark that this is a specific string here,

```suggestion
unspecified or set to ``io.cilium/bgp-control-plane`` to be announced by a policy.
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks for the addition!
I've got a few comments.

* The beginning of the subsection says:
  > Virtual routers will announce the ingress IPs of any LoadBalancer services that matches the ``.serviceSelector`` of the virtual router.

  But then this is not correct, it should be `any ... without loadBalancerClass`? I'm a bit worried readers might read the beginning but skip the restriction you're mentioning. On the other hand, the subsection is small, so if users don't read all of it there's only so much we can do I guess?

* Should the paragraph briefly explain why service with other `loadBalancerClass`es are not announced?

File: Documentation/network/bgp-control-plane.rst (hunk 1 above)
Comment (danehans) [reply Documentation/network/bgp-control-plane.rst]: Commit 0c5f501 resolves this comment.

**Comment by danehans** [comment]
@qmonnet thank you for reviewing my PR.

> I'm a bit worried readers might read the beginning but skip the restriction you're mentioning. 

I have repositioned and reworded the documentation to address this concern, PTAL.

> Should the paragraph briefly explain why service with other loadBalancerClasses are not announced?

IMHO the link to the k8s lb class doc provides the reader with the opportunity to learn more. The current approach provides a balance of directness with the opportunity for the reader to learn more through the link. I don't have a strong opinion on this issue so I'm happy to update if you and others feel differently.

**Review by qmonnet — APPROVED** [review]
OK, no strong opinion either, and I'm good with the current version. Thanks a lot!

**Pull request merged**